crates/modelardb_manager/src/metadata.rs

```diff
@@ -20,38 +20,42 @@ use std::str::FromStr;
 
 use futures::TryStreamExt;
 use modelardb_common::errors::ModelarDbError;
-use modelardb_common::metadata;
-use modelardb_common::metadata::model_table_metadata::ModelTableMetadata;
+use modelardb_common::metadata::{try_new_postgres_table_metadata_manager, TableMetadataManager};
 use modelardb_common::types::ServerMode;
-use sqlx::{Executor, PgPool, Row};
+use sqlx::{PgPool, Postgres, Row};
 use uuid::Uuid;
 
 use crate::cluster::Node;
 
 /// Stores the metadata required for reading from and writing to the tables and model tables and
 /// persisting edges. The data that needs to be persisted is stored in the metadata database.
-#[derive(Clone)]
 pub struct MetadataManager {
     /// Pool of connections to the metadata database.
     metadata_database_pool: PgPool,
+    /// Metadata manager used to interface with the subset of the manager metadata database related
+    /// to tables and model tables.
+    pub(crate) table_metadata_manager: TableMetadataManager<Postgres>,
```

Comment:
It is public since we no longer provide methods in the manager metadata manager to access the table metadata manager. It is just used directly instead, for example: 
```rust
self.context
    .metadata_manager
    .table_metadata_manager
    .save_table_metadata(&table_name, sql)
```

I think this is better since it removes a lot of boilerplate code and the distinction between table metadata functionality and manager metadata functionality is quite clear. I have left the doc comment as it was. The next PR will use `table_metadata_manager` more so i will reconsider then whether to keep it public.